pandas/_libs/tslibs/src/datetime/np_datetime_strings.c

```diff
@@ -69,7 +69,8 @@ This file implements string parsing and creation for NumPy datetime.
 int parse_iso_8601_datetime(const char *str, int len, int want_exc,
                             npy_datetimestruct *out,
                             NPY_DATETIMEUNIT *out_bestunit,
-                            int *out_local, int *out_tzoffset) {
+                            int *out_local, int *out_tzoffset,
+                            const char* format, int format_len, int exact) {
```

Comment:
Do we ever have to worry about `format_len` being negative? Seems like it could cause undefined behavior if that was passed this way - would be best to validate the argument and set an appropriate error up front

Reply:
That shouldn't ever happen - it's the same with the `len` variable, it comes from

https://github.com/pandas-dev/pandas/blob/f3060c9afab7f8cd8e4362324cc3978d71314a73/pandas/_libs/tslibs/np_datetime.pyx#L284-L296

so I don't see how `len` nor `format_len` could be negative, as `parse_iso_8601_datetime` is only used internally

I've added a check anyway